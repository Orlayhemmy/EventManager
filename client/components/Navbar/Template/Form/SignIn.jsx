import React from 'react';

export default class SignIn extends React.Component {
  render() {
    const {
      signinError,
      signinSubmit,
      loginEmail,
      loginPassword,
      errorEmail,
      errorPass,
      onChange,
      auth,
      password
    } = this.props.navbarProps;
    return (
      <form class="px-4 py-3" onSubmit={signinSubmit}>
        <div className="form-group">
          <span className="help-block">
            {errorEmail || auth.signinError}
          </span>
          <label for="email">Email Address</label>
          <input
            id="loginEmail"
            value={loginEmail}
            placeholder="Email Address"
            type="email"
            className="form-control"
            error={errorEmail}
            onChange={onChange}
            required
          />
        </div>
        <div className="form-group">
          <span className="help-block">{password}</span>
          <label for="email">Password</label>
          <input
            id="loginPassword"
            value={loginPassword}
            placeholder="123456789"
            type="password"
            className="form-control"
            error={errorPass}
            onChange={onChange}
            required
          />
        </div>
        <button type="submit" class="btn btn-primary">
          Sign in
        </button>
      </form>
    )
  }
} 