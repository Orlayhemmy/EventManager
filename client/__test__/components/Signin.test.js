import React from 'react';
import SignIn from '../../components/Navbar/Template/Form/SignIn';

describe('Signin', () => {
  const navbarProps = {
    signinError: '',
    signinSubmit: () => {},
    loginEmail: '',
    loginPassword: '',
    errorEmail: '',
    errorPass: '',
    onChange: () => {},
    auth: { signinError: '' },
    password: ''
  };
  const wrapper = shallow(<SignIn navbarProps={navbarProps} />);
  it('should render the component without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
