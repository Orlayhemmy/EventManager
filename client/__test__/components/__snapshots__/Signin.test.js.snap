// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image upload should render the component without errors 1`] = `
<form
  className="px-4 py-3"
  onSubmit={[Function]}
>
  <span
    className="help-block"
  />
  <div
    className="form-group"
  >
    <label
      htmlFor="email"
    >
      Email Address
    </label>
    <input
      className="form-control"
      error=""
      id="loginEmail"
      onChange={[Function]}
      placeholder="Email Address"
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <span
      className="help-block"
    />
    <label
      htmlFor="email"
    >
      Password
    </label>
    <input
      className="form-control"
      error=""
      id="loginPassword"
      onChange={[Function]}
      placeholder="123456789"
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary mb-3"
    type="submit"
  >
    Sign in
  </button>
  <br />
  <span>
    <Link
      replace={false}
      to="/recover-password"
    >
      Forgot Password? Click Here
    </Link>
  </span>
</form>
`;
